pkg/controller/batchrelease/control/util.go

```diff
@@ -63,6 +65,42 @@ func IsControlledByBatchRelease(release *v1beta1.BatchRelease, object client.Obj
 	return false
 }
 
+// only when IsReadyForBlueGreenRelease returns true, can we go on to the next batch
+func ValidateReadyForBlueGreenRelease(object client.Object) error {
+	// check the annotation
+	if object.GetAnnotations()[util.BatchReleaseControlAnnotation] == "" {
+		return fmt.Errorf("workload has no control info annotation")
+	}
+	switch o := object.(type) {
+	case *apps.Deployment:
+		// must be RollingUpdate
+		if len(o.Spec.Strategy.Type) > 0 && o.Spec.Strategy.Type != apps.RollingUpdateDeploymentStrategyType {
+			return fmt.Errorf("deployment strategy type is not RollingUpdate")
+		}
+		if o.Spec.Strategy.RollingUpdate == nil {
+			return fmt.Errorf("deployment strategy rollingUpdate is nil")
+		}
+		// MinReadySeconds and ProgressDeadlineSeconds must be set
+		if o.Spec.MinReadySeconds != v1beta1.MaxReadySeconds || o.Spec.ProgressDeadlineSeconds == nil || *o.Spec.ProgressDeadlineSeconds != v1beta1.MaxProgressSeconds {
+			return fmt.Errorf("deployment strategy minReadySeconds or progressDeadlineSeconds is not MaxReadySeconds or MaxProgressSeconds")
+		}
+
+	case *appsv1alpha1.CloneSet:
+		// must be ReCreate
+		if len(o.Spec.UpdateStrategy.Type) > 0 && o.Spec.UpdateStrategy.Type != appsv1alpha1.RecreateCloneSetUpdateStrategyType {
+			return fmt.Errorf("cloneSet strategy type is not ReCreate")
+		}
+		// MinReadySeconds and ProgressDeadlineSeconds must be set
+		if o.Spec.MinReadySeconds != v1beta1.MaxReadySeconds {
+			return fmt.Errorf("cloneSet strategy minReadySeconds is not MaxReadySeconds")
+		}
+
+	default:
+		panic("unsupported workload type to ValidateReadyForBlueGreenRelease function")
```

Comment:
can we validate the workload type and rollout version in validating webhook of rollout

Reply:
done

Reply:
ps. the v1alpha1 Rollout doesn't has bluegreen field from the begining